### ISSUE TITLE
Make `ApplicationLoader.load` asynchronously.

### ISSUE THREAD

**Issue by jilen** [playframework/playframework#10834]
If `load` involve some blocking,  underlying thread might be parked, and if all other threads in default pool waiting for loading app at same time. Then the loading thread will  be parked forever and app will never get started.

```
"application-akka.actor.default-dispatcher-6" prio=5 Id=27 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-7" prio=5 Id=36 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-8" prio=5 Id=37 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-9" prio=5 Id=38 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-10" prio=5 Id=50 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-11" prio=5 Id=51 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-12" prio=5 Id=52 WAITING
"application-akka.actor.default-dispatcher-13" prio=5 Id=53 BLOCKED on CustomLoader@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
```

Another possible  fix is loading app in an isolated pool other than the app pool.
